src/controller_examples/rabbitmq_controller/proof/helper_invariants/proof.rs

```diff
@@ -0,0 +1,1359 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: MIT
+#![allow(unused_imports)]
+use super::predicate::*;
+use crate::kubernetes_api_objects::{
+    api_method::*, common::*, error::*, owner_reference::*, prelude::*, resource::*,
+};
+use crate::kubernetes_cluster::spec::{
+    cluster::*,
+    cluster_state_machine::Step,
+    controller::common::{ControllerActionInput, ControllerStep},
+    message::*,
+};
+use crate::rabbitmq_controller::{
+    common::*,
+    proof::{predicate::*, resource::*},
+    spec::types::*,
+};
+use crate::temporal_logic::{defs::*, rules::*};
+use crate::vstd_ext::{multiset_lib, seq_lib, string_view::*};
+use vstd::{multiset::*, prelude::*, string::*};
+
+verus! {
+
+pub proof fn lemma_always_cr_objects_in_etcd_satisfy_state_validation(spec: TempPred<RMQCluster>)
```

Comment:
Seems to me this can be made generic to all controllers?

Reply:
No

Reply:
Can you remind me on why?